seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/CoordinatorService.java

```diff
@@ -340,23 +338,22 @@ public PassiveCompletableFuture<Void> submitJob(long jobId, Data jobImmutableInf
             try {
                 runningJobInfoIMap.put(jobId, new JobInfo(System.currentTimeMillis(), jobImmutableInformation));
                 runningJobMasterMap.put(jobId, jobMaster);
-                jobMaster.init(runningJobInfoIMap.get(jobId).getInitializationTimestamp());
-                jobMaster.initCheckPointManager();
+                jobMaster.init(runningJobInfoIMap.get(jobId).getInitializationTimestamp(), false);
+                // We specify that when init is complete, the submitJob is complete
+                jobSubmitFuture.complete(null);
             } catch (Throwable e) {
                 logger.severe(String.format("submit job %s error %s ", jobId, ExceptionUtils.getMessage(e)));
-                voidCompletableFuture.completeExceptionally(e);
-            } finally {
-                // We specify that when init is complete, the submitJob is complete
-                voidCompletableFuture.complete(null);
+                jobSubmitFuture.completeExceptionally(e);
             }
-
-            try {
-                jobMaster.run();
-            } finally {
-                onJobDone(jobMaster, jobId);
+            if (!jobSubmitFuture.isCompletedExceptionally()) {
+                try {
+                    jobMaster.run();
+                } finally {
+                    onJobDone(jobMaster, jobId);
```

Comment:
`onJobDone()` will not be call when `jobSubmitFuture.isCompletedExceptionally() == true`. I think we need remove the job from `runningJobInfoIMap` when the job submit failed.